src/full/Agda/Syntax/Translation/ConcreteToAbstract.hs

```diff
@@ -868,8 +868,14 @@ instance ToAbstract C.Expr where
         i       = ExprRange r
         alit    = A.Lit i l
         mkApp q = A.App (defaultAppInfo r) (A.Def q) . defaultNamedArg
+
+        -- #4925: Require fromNat/fromNeg to be in scope *unqualified* for literal overloading to
+        -- apply.
         ensureInScope :: Maybe I.Term -> ScopeM (Maybe I.Term)
-        ensureInScope v@(Just (I.Def q _)) = ifM (isNameInScope q <$> getScope) (return v) (return Nothing)
+        ensureInScope v@(Just (I.Def q _)) = do
+          inverseScopeLookupName' AmbiguousNothing q <$> getScope >>= \case
+            C.QName{} : _ -> return v -- NOTE: inverseScopeLookupName puts unqualified names first
```

Comment:
The success of type-checking relies on a particular behavior of `inverseScopeLookupName'`... my stomach finds this a bit brittle.  Would feel better if this would be packaged as a dedicated method in `Scope.Base`, something like `isNameInScopeUnqualified`.

Reply:
In this case the behaviour is part of the specification of `inverseScopeLookupName'`, but indeed wrapping it in a function would be nicer.